### PR TITLE
feat: forge-inspect script for contracts-core

### PR DESCRIPTION
**Pull request by odyslam**
> It's nifty!

### CODE REVIEW

File: packages/contracts-core/scripts/storage-inspect.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+dir=$(dirname "$0")
+
+echo "Creating storage layout diagrams.."
+
+echo "=======================" > $dir/../.storage-layout
+echo "👁👁 STORAGE LAYOUT snapshot 👁👁" >> $dir/../.storage-layout
+echo "=======================" >> $dir/../.storage-layout
+echo -e "\nCurrent git commit $(git rev-parse HEAD) \n\n" >> $dir/../.storage-layout
```
[the remark this answers is not in the document]
Comment (prestwich) [reply packages/contracts-core/scripts/storage-inspect.sh:27]: ya, the changes are already associated with a commit. Let git manage exposing that info

File: packages/contracts-core/package.json [hunk 2]
```diff
@@ -31,7 +31,8 @@
     "link:node_modules": "ln -s ../../node_modules",
     "snapshot": "forge clean && FOUNDRY_PROFILE=core forge snapshot",
     "snapshot:check": "forge clean && FOUNDRY_PROFILE=core forge snapshot --check",
-    "gen-proof": "../../scripts/accumulator-cli"
+    "gen-proof": "../../scripts/accumulator-cli",
+    "gen-sl": "scripts/storage-inspect.sh"
```
Comment (prestwich) [inline packages/contracts-core/package.json:35]: this should be added to the `build` script

**Review by prestwich — CHANGES_REQUESTED** [review]
add a CI step that generates these and fails CI if it generates a different file

**Comment by odyslam** [comment]
Yeah. Let me add the bridge contracts as well and finally make an addition to the GitHub actions? 

We could have something like what was mentioned above. The test will fail if the storage has changed so the engineer has to explicitly acknowledge the change, create and commit the new storage layout so that the test doesn't fail, and then merge. What do you say?

**Comment by odyslam** [comment]
Closing in favour of #403

**Pull request closed without merging**